gui/src-tauri/src/main.rs

```diff
@@ -306,7 +305,23 @@ fn valid_java_paths() -> Vec<(OsString, i32)> {
 
 	// Otherwise check if anything else is a supported version
 	let mut childs = vec![];
-	for java in which_all(JAVA_BIN).unwrap() {
+	cfg_if::cfg_if! {
+		if #[cfg(target_os = "macos")] {
```

Comment:
Use `cfg!()` bool instead of feature gating, to avoid complexity of nesting introduced by `cfg_if` as well as avoid separate code paths. The compiler is going to treat the `if true {}` as a const anyway and compile out the dead branch anyway.

Reply:
just tried it, not worth it. Different typings because one of them uses a filter_map which means it has the Filter type, and another one has some unknown type that impl's Iterator

Reply:
I would just make them collect into a vector then iterate on the vector but like...who really cares anyway. Resolving.